### PR TITLE
Remove event manager base

### PR DESCRIPTION
**Pull request by TimvdLippe**
This manager was actually a bunch of static state of `WatchDogEvent`. As such, refactor the class away and replace it with static state. Moreover, move the switch-case logic into the enum type, to make sure new intervals adhere to this contract. (I will need this when I am going to add new intervals for the SA perspectives)

Note that the current solution is still not perfect, as the static state is annoying to work with, however I think it is at least an improvement over the previous situation. I actually found out some dependency patterns between types which would recursively call `update`.

The tests are very flaky, as they rely on `sleep`. I have verified that the test suite passes on my local computer, but they fail most of the time as well. I do not have time to refactor this test suite as well, so this could be a TODO.

This is part of #275

### CODE REVIEW

**Comment by TimvdLippe** [comment]
Kind of expected, but the two Travis builds failed on different tests. Moreover, locally they were passing and debugging shows that they usually have the correct values. Not sure how to tackle this flakiness :cry:

**Comment by TimvdLippe** [comment]
In #278 the builds also failed, in which both `testNoMoreAdditionalUserActivitiesShouldNotCloseReading` failed. This is really weird, as there are multiple tests that depend on this exact test (they call it as prerequisite). However, none of these tests failed. Thus it is probably related to flakiness in the sleeps. I have added back a sleep to hopefully deflake this test, but I fear that it will remain flaky until we can deterministically handle the time constraints.

**Comment by Inventitech** [comment]
Please factor out duplication and move out  public enum EventType

**Comment by TimvdLippe** [comment]
@Inventitech Updated :tada: I eventually chose to use an enum for the activity notifiers again, as that was easier than to create the methods on the initialization manager (as these are IDE-specific). This also opens up the opportunity to mock the notifiers later to get rid of the sleeps and hopefully deflake the tests :smile:

File: core/src/nl/tudelft/watchdog/core/logic/ui/events/WatchDogEventType.java [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+            intervalManager.addInterval(new IDEOpenInterval(forcedDate));
+            InActivityNotifiers.USER_INACTIVITY.trigger(forcedDate);
+        }
+    }, END_IDE {
+        @Override
+        public void process(Date forcedDate, Object source) {
+            InActivityNotifiers.USER_INACTIVITY.cancelTimer(forcedDate);
+        }
+    },
+
```
Comment (Inventitech) [inline core/src/nl/tudelft/watchdog/core/logic/ui/events/WatchDogEventType.java]: Can you separate the enum types like this for readability throughout the document? Thanks!

Comment (Inventitech) [reply core/src/nl/tudelft/watchdog/core/logic/ui/events/WatchDogEventType.java]: Maybe adding comments to understand the separation of the different subtypes helps

File: core/src/nl/tudelft/watchdog/core/util/WatchDogUtilsBase.java [hunk 2]
```diff
@@ -75,7 +75,7 @@ public static long countSLOC(String text) {
 	 * @return Whether the string with white spaces trimmed is empty.
 	 */
 	public static boolean isEmptyOrHasOnlyWhitespaces(String string) {
-		return isEmpty(string) ? true : string.trim().isEmpty();
+		return isEmpty(string) || string.trim().isEmpty();
```
Comment (Inventitech) [inline core/src/nl/tudelft/watchdog/core/util/WatchDogUtilsBase.java]: nice :)

File: eclipse/plugin/src/nl/tudelft/watchdog/eclipse/logic/ui/WatchDogEventManager.java [hunk 3]
```diff
@@ -1,56 +0,0 @@
...
-import nl.tudelft.watchdog.core.logic.interval.IDEIntervalManagerBase;
-import nl.tudelft.watchdog.core.logic.interval.intervaltypes.PerspectiveInterval;
-import nl.tudelft.watchdog.core.logic.ui.WatchDogEventManagerBase;
-import nl.tudelft.watchdog.core.logic.ui.events.WatchDogEvent;
-import nl.tudelft.watchdog.core.logic.ui.events.WatchDogEvent.EventType;
-import nl.tudelft.watchdog.eclipse.logic.document.DocumentCreator;
-import nl.tudelft.watchdog.eclipse.logic.document.EditorWrapper;
-import nl.tudelft.watchdog.eclipse.logic.interval.intervaltypes.JUnitInterval;
-
-/**
```
Comment (Inventitech) [inline eclipse/plugin/src/nl/tudelft/watchdog/eclipse/logic/ui/WatchDogEventManager.java]: Can we preserve this comment somehow (I think it best fits in the process method)?

File: core/src/nl/tudelft/watchdog/core/logic/ui/InActivityNotifiers.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
+package nl.tudelft.watchdog.core.logic.ui;
+
+import nl.tudelft.watchdog.core.logic.ui.events.WatchDogEventType;
+import nl.tudelft.watchdog.core.util.WatchDogGlobals;
+
+import java.util.Date;
+
+public enum InActivityNotifiers {
```
[the remark this answers is not in the document]
Comment (TimvdLippe) [reply core/src/nl/tudelft/watchdog/core/logic/ui/InActivityNotifiers.java]: Done :+1:

File: core/src/nl/tudelft/watchdog/core/logic/ui/events/WatchDogEventType.java (hunk 1 above)
Comment (TimvdLippe) [reply core/src/nl/tudelft/watchdog/core/logic/ui/events/WatchDogEventType.java]: I have separated the enum types with one new line between each type

File: eclipse/plugin/src/nl/tudelft/watchdog/eclipse/logic/ui/WatchDogEventManager.java (hunk 3 above)
Comment (TimvdLippe) [reply eclipse/plugin/src/nl/tudelft/watchdog/eclipse/logic/ui/WatchDogEventManager.java]: Done :+1:

File: eclipse/tests/src/nl/tudelft/watchdog/logic/ui/InactivityNotifierTest.java [hunk 5]
```diff
@@ -1,98 +0,0 @@
...
-import nl.tudelft.watchdog.core.logic.ui.events.WatchDogEvent;
-import nl.tudelft.watchdog.core.logic.ui.events.WatchDogEvent.EventType;
-import nl.tudelft.watchdog.eclipse.logic.ui.WatchDogEventManager;
-import nl.tudelft.watchdog.eclipse.util.WatchDogUtils;
-
-/**
- * Tests for inactivity notifier. These tests have the potential to flicker, as
- * they deal with precise timing.
- */
-public class InactivityNotifierTest {
```
[the remark this answers is not in the document]
Comment (TimvdLippe) [reply eclipse/tests/src/nl/tudelft/watchdog/logic/ui/InactivityNotifierTest.java]: Oops yes, lost in the refactoring. Added back :+1:

File: core/src/nl/tudelft/watchdog/core/logic/ui/TimeSynchronityChecker.java [hunk 6]
```diff
@@ -66,9 +61,8 @@ public void run() {
 					intervalManager.closeAllIntervals(new Date(
 							previousExecutionDate + UPDATE_RATE));
 					intervalManager.generateAndSetSessionSeed();
-					eventManager.update(new WatchDogEvent(this,
-							EventType.START_IDE));					
-					eventManager.updatePerspectiveInterval();
+                    WatchDogEventType.START_IDE.process(this);
+                    WatchDogEventType.editorSpecificImplementation.updatePerspectiveInterval();
```
[the remark this answers is not in the document]
Comment (TimvdLippe) [reply core/src/nl/tudelft/watchdog/core/logic/ui/TimeSynchronityChecker.java]: Do you have a proposal for a different way to implement this which you find not ugly?

[the remark this answers is not in the document]
Comment (Inventitech) [reply core/src/nl/tudelft/watchdog/core/logic/ui/TimeSynchronityChecker.java]: I wonder, with the rename of update into process in the other function, wouldn't it make more sense to call this processPersepctiveInterval()?

**Comment by Inventitech** [comment]
Almost good to go.

File: core/src/nl/tudelft/watchdog/core/logic/ui/TimeSynchronityChecker.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (TimvdLippe) [reply core/src/nl/tudelft/watchdog/core/logic/ui/TimeSynchronityChecker.java]: Ah yes, done :+1:

**Pull request merged**